bip-tombriar-compressed-transactions.mediawiki

```diff
@@ -0,0 +1,268 @@
+|-
+| Standard Hash || 1 Bit || A flag to determine if this Input's Signature Hash Type is standard (0x00 for Taproot, 0x01 for Legacy/Segwit).
+|-
+| Standard Squence || 2 Bits || A CompactSize flag for the inputs sequence. Encode literal values as follows: 1 = 0x00000000, 2 = 0xFFFFFFFE, 3 = 0xFFFFFFFF.
+|}
+
+<span id="output-metadata"></span>
+==== Output Metadata ====
+
+{| class="wikitable" style="margin:auto"
+|-
+! Name !! Width !! Description
+|-
+| Encoded Script Type || 3 Bits || [[#script-type-encoding|Encoded Script Type]].
+|}
+
+<span id="script-type-encoding"></span>
+==== Script Type Encoding ====
+
+{| class="wikitable" style="margin:auto"
+|-
+! Script Type !! Value
+|-
+| Uncompressed P2PK || 0b000
+|-
+| Compressed P2PK || 0b001
+|-
+| P2PKH || 0b010
+|-
+| P2SH || 0b011
+|-
+| P2WSH || 0b100
+|-
+| P2WPKH || 0b101
+|-
+| P2TR || 0b110
+|-
+| Uncompressed Custom Script || 0b111
+|}
+
+<span id="input-data"></span>
+==== Input Data ====
+
+{| class="wikitable" style="margin:auto"
+|-
+! Name !! Width !! Description
+|-
+| Sequence || 0-5 Bytes || An Optional VarInt containing the sequence if it was non-standard.
+|-
+| Txid Blockheight || 1-5 Bytes || A VarInt Either containing 0 if this an uncompressed input, or it contains the offset from Minimum Blockheight for this Txid.
+|-
+| Txid/Signature Data || 65+ Bytes || Txid/Signatures are determined to be uncompressed either by the output script of the previous transaction, or if the Txid Blockheight is zero. For each Compressed Txid/Signature see [[#compressed-data|here]]. For each Uncompressed Txid/Signature see [[#uncompressed-data|here]].
+|}
+
+<span id="compressed-data"></span>
+==== Compressed Txid/Signature Data ====
+
+{| class="wikitable" style="margin:auto"
+|-
+! Name !! Width !! Description
+|-
+| Txid Block Index || 1-5 Bytes || A VarInt containing the flattened index from the Txid Blockheight for the Vout.
+|-
+| Signature || 64 Bytes || Contains the 64 byte signature.
+|-
+| Hash Type || 0-1 Bytes || An Optional Byte containing the Hash Type if it was non-standard.
+|}
+
+<span id="uncompressed-data"></span>
+==== Uncompressed Txid/Signature Data ====
+
+{| class="wikitable" style="margin:auto"
+|-
+! Name !! Width !! Description
+|-
+| Txid || 32 Bytes || Contains the 32 byte Txid.
+|-
+| Vout || 1-5 Bytes || A CompactSize Containing the Vout of the Txid.
+|-
+| Signature || 2+ Bytes || A VLP-Bytestream containing the signature.
+|}
+
+<span id="output-data"></span>
+==== Output Data ====
+
+{| class="wikitable" style="margin:auto"
+|-
+! Name !! Width !! Description
+|-
+| Output Script || 2+ Bytes || A VLP-Bytestream containing the output script.
+|-
+| Amount || 1-9 Bytes || A VarInt containing the output amount.
+|}
+
+==== Ideal Transaction ====
+
+The target transaction for the most optimal compression was chosen
+based off the most common transactions that are likely to be used
+for purposes that requires the best compression.
```

Comment:
I'd reword this because it's hard to tell whether you're about to describe an "ideal" transaction or a "common" one or what. I suggest

```
The compression scheme was designed to be optimal for a "typical" transaction spending a few close-in-age inputs and having one or two outputs. Here are size values for such a transaction which demonstrate the effectiveness of the compression.
```

Reply:
Fixed, Thanks!